src/data/lock.ts

```diff
@@ -30,15 +31,20 @@ export const callProtectedLockService = async (
   service: ProtectedLockService
 ) => {
   let code: string | undefined;
+  const lockRegistryEntry = await getExtendedEntityRegistryEntry(
+    hass,
+    stateObj.entity_id
+  );
+  const defaultCode = lockRegistryEntry.options?.lock?.default_code;
```

Comment:
```suggestion
  const lockRegistryEntry = await getExtendedEntityRegistryEntry(
    hass,
    stateObj.entity_id
  ).catch(() => undefined);
  const defaultCode = lockRegistryEntry?.options?.lock?.default_code;
```
The promise can fail if the lock isn't in the entity registry (demo lock for example)